.evergreen.yaml

```diff
@@ -68,12 +74,31 @@ functions:
       params:
         ignore_missing_file: true
         file: src/atlas-service-broker/version_expansion.yaml
+
   "build_binaries":
     - command: shell.exec
       params:
         working_dir: src/atlas-service-broker
         script: |
           ./dev/scripts/build-production-binary.sh artifacts/atlas-service-broker-linux-arm64-${version}
+
+  # Upload integration test binary to S3. MMS will use this binary as part of the Atlas test suite.
+  "upload_test_binary":
+    - command: shell.exec
+      params:
+        working_dir: src/atlas-service-broker
+        script: |
+          go test -v -timeout 1h -c -o artifacts/int-tests ./test/integration
+    - command: s3.put
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        local_file: src/atlas-service-broker/artifacts/int-tests
+        remote_file: atlas-service-broker/artifacts/int-tests
```

Comment:
can you give this filename something that indicates what version of the AOSB it is? 

A good way to do this might be: 

atlas-service-broker/artifacts/<githash>/int-tests

Then also have a metadata file: atlas-service-broker/builds.json 
which links to the latest builds

```
{ 
   "master": { "int-test-build": "<url>", "githash": "<sha>" } 
}
```
or anything else which might be useful

Reply:
^ then the mms test suite can grab the metadata file & then download the right binary

Reply:
As this upload is only run as part of release we could name it after the version number perhaps? I'll try!

Reply:
gitsha would be better, as they are immutable. 

Ideally, there would be a git tag with the customer-visible version number attached to the git sha

Reply:
Agreed—though ideally our version tags are treated as immutable as well.

Reply:
Got it! I've added a new meta file for builds and namespaced the artifacts by hash. Have another look!